Bottom: 16c37bdabbc39208ce914055452f4259fe2637f6
Top:    f0c763aaae79a2cfdaa9da43ae738be42623c03b
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-09-25 14:04:52 -0500

fixup buildbot


---
diff --git a/ansible-playbook/buildbot.yml b/ansible-playbook/buildbot.yml
index e019df238..1ea46fb01 100644
--- a/ansible-playbook/buildbot.yml
+++ b/ansible-playbook/buildbot.yml
@@ -13,7 +13,7 @@
         name: nexus
         groups: services
         ip_addr: "{{ nexus_ip_addr }}"
-        uri: "http://{{nexus_hostname}}/repository"
+        uri: "repository"
     - name: Scan Gitlab ssh keys
       command: ssh-keyscan {{gitlab_hostname}}
       register: gitlab_keys
@@ -22,7 +22,12 @@
         name: buildbot-master
       vars:
         known_hosts: "{{gitlab_keys.stdout}}"
-        asset_host: "{{hostvars['nexus']['uri']}}"
+        ntel_release_uri:    "https://{{nexus_hostname}}/repository/ntel-release"
+        ntel_beta_uri:       "https://{{nexus_hostname}}/repository/ntel-unofficial"
+        ntel_sstate_uri:     "https://{{nexus_hostname}}/repository/ntel-sstate"
+        ptxdist_release_uri: "https://{{nexus_hostname}}/repository/ptxdist-release"
+        ptxdist_beta_uri:    "https://{{nexus_hostname}}/repository/ptxdist-unofficial"
+
     - name: Add buildbot user to Gitlab
       import_role:
         name: gitlab-user
@@ -43,13 +48,21 @@
     - buildbot-worker-ntel
     - buildbot-worker-ptxdist
   vars:
-    asset_host: "{{hostvars['nexus']['uri']}}"
+    asset_host: "https://{{nexus_hostname}}/repository"
     asset_user: "buildbot"
     asset_password: "{{nexus_buildbot_password}}"
     buildbot_master: "{{hostvars[buildbot_hostname].ip_addr}}"
     container_etc_hosts:
       nexus.novatech-llc.com: "{{hostvars['nexus']['ip_addr']}}"
     known_hosts: "{{ gitlab_keys.stdout }}"
+    asset_downloads:   "{{ asset_host }}/download"
+    asset_orphans:     "{{ asset_host }}/orphans"
+    asset_debian:      "{{ asset_host }}/debian"
+    asset_sourceforge: "{{ asset_host }}/sourceforge"
+    asset_gnu:         "{{ asset_host }}/gnu"
+    asset_xorg:        "{{ asset_host }}/xorg"
+    asset_kernel:      "{{ asset_host }}/kernel"
+    asset_pypi:        "{{ asset_host }}/pypi"
   tasks:
     - name: Scan Gitlab ssh keys
       command: ssh-keyscan {{gitlab_hostname}}
diff --git a/roles/buildbot-master/defaults/main.yml b/roles/buildbot-master/defaults/main.yml
index d23f6eee7..efeb94099 100644
--- a/roles/buildbot-master/defaults/main.yml
+++ b/roles/buildbot-master/defaults/main.yml
@@ -35,7 +35,11 @@ config_hostdir: '{{ docker_restore_config_base_dir }}/{{ container_name }}'
 
 # environment passed to the container
 container_env:
-  ASSET_HOST: '{{asset_host}}'
+  NTEL_BETA_URI: '{{ ntel_beta_uri }}'
+  NTEL_RELEASE_URI: '{{ ntel_release_uri }}'
+  NTEL_SSTATE_URI: '{{ ntel_sstate_uri }}'
+  PTXDIST_BETA_URI: '{{ ptxdist_beta_uri }}'
+  PTXDIST_RELEASE_URI: '{{ ptxdist_release_uri }}'
   BUILDBOT_DATABASE: 'sqlite:///{{ data_path }}/state.sqlite'
   BUILDBOT_KNOWN_HOSTS_FILE: '{{secrets_path}}/{{secrets_known_hosts}}'
   BUILDBOT_IRC_CHANNELS: "{{ buildbot_irc_channels | join(',') }}"
diff --git a/roles/buildbot-master/files/openembedded.py b/roles/buildbot-master/files/openembedded.py
index a2a59480a..72f8ed7e2 100755
--- a/roles/buildbot-master/files/openembedded.py
+++ b/roles/buildbot-master/files/openembedded.py
@@ -12,7 +12,10 @@ DEFAULT_BBFLAGS = '-k'
 DEFAULT_BRANCH = 'morty'
 DEFAULT_CODEBASE = "ntel/setup-scripts"
 DEFAULT_REPO = 'git@git.novatech-llc.com:ntel/setup-scripts.git'
-ASSET_HOST = os.getenv("ASSET_HOST", default="http://127.0.0.1")
+
+BETA_URI = os.getenv("NTEL_BETA_URI", default="http://127.0.0.1")
+RELEASE_URI = os.getenv("NTEL_RELEASE_URI", default="http://127.0.0.1")
+SSTATE_URI = os.getenv("NTEL_SSTATE_URI", default="http://127.0.0.1")
 
 NTEL_LAYERS = {
     DEFAULT_CODEBASE: {
@@ -201,16 +204,6 @@ def ComputeBuildProperties(props):
     if not version:
         version = newprops['timestamp']
 
-    urlpath = 'unofficial'
-
-    if props.getProperty('release_pin'):
-        urlpath = 'release'
-
-    if props.getProperty('scheduler') == 'ntel-nightly':
-        urlpath = 'nightly'
-
-    urlpath = os.path.join('ntel', urlpath, version)
-
     newprops['artifacts'] = {}
     machines = [props.getProperty('machine')]
 
@@ -234,8 +227,14 @@ def ComputeBuildProperties(props):
             a['prefix'],
         )
 
-        a['url'] = "%s/%s/%s" % (
-            ASSET_HOST,
+        if props.getProperty('release_pin'):
+            uri = RELEASE_URI
+        else:
+            uri = BETA_URI
+
+        a['url'] = os.path.join(
+            uri,
+            version,
             urlpath,
             a['artifact'],
         )
diff --git a/roles/buildbot-master/files/ptxdist.py b/roles/buildbot-master/files/ptxdist.py
index aab8d2317..304704c6f 100644
--- a/roles/buildbot-master/files/ptxdist.py
+++ b/roles/buildbot-master/files/ptxdist.py
@@ -8,7 +8,9 @@ c = WorkerConfig = {}
 
 DEFAULT_BRANCH = 'master'
 DEFAULT_REPO = 'git@git.novatech-llc.com:Orion-ptxdist/workspace-ptxdist2'
-ASSET_HOST = os.getenv("ASSET_HOST", default="http://127.0.0.1")
+
+BETA_URI = os.getenv("PTXDIST_BETA_URI", default="http://127.0.0.1")
+RELEASE_URI = os.getenv("PTXDIST_RELEASE_URI", default="http://127.0.0.1")
 
 collections = {
     "armeb-xscale": "armeb-base",
@@ -311,32 +313,20 @@ def ComputeBuildProperties(props):
     )
 
     if props.getProperty('release'):
-        urlpath = os.path.join(
-            'ptxdist',
-            'release',
-            version
-        )
-    elif props.getProperty('scheduler') == "ptxdist-nightly":
-        urlpath = os.path.join(
-            'ptxdist',
-            'nightly',
-            version
-        )
+        uri = RELEASE_URI
     else:
-        urlpath = os.path.join(
-            'ptxdist',
-            'beta',
-            version
-        )
+        uri = BETA_URI
 
     newprops['ipkg_url'] = os.path.join(
-        ASSET_HOST,
+        uri,
+        version,
         urlpath,
         ipkg_artifact,
     )
 
     newprops['image_url'] = os.path.join(
-        ASSET_HOST,
+        uri,
+        version,
         urlpath,
         image_artifact,
     )
diff --git a/roles/buildbot-worker-ntel/defaults/main.yml b/roles/buildbot-worker-ntel/defaults/main.yml
index c6097fe56..ae729ce14 100644
--- a/roles/buildbot-worker-ntel/defaults/main.yml
+++ b/roles/buildbot-worker-ntel/defaults/main.yml
@@ -2,7 +2,7 @@
 # defaults file for buildbot-worker-ntel
 
 # host containing build assets
-asset_host: 'http://{{ansible_facts.default_ipv4.address}}/'
+asset_downloads: 'http://{{ansible_facts.default_ipv4.address}}/'
 
 # address of buildbot master service
 buildbot_master: '{{ansible_facts.default_ipv4.address}}'
@@ -27,7 +27,6 @@ cache_volume: '{{ docker_name_prefix }}{{ container_hostname }}-cache'
 
 # environment passed to the container
 container_env:
-  ASSET_HOST: '{{asset_host}}/ntel'
   BUILDBOT_KNOWN_HOSTS_FILE: '{{secrets_path}}/{{secrets_known_hosts}}'
   BUILDBOT_MASTER: '{{ buildbot_master }}'
   BUILDBOT_WORKER_PORT: '{{ buildbot_worker_port }}'
@@ -72,7 +71,7 @@ hsm_host: '{{ansible_facts.default_ipv4.address}}'
 image_args:
   BUILDBOT_UID: '{{buildbot_uid}}'
   BUILDBOT_VERSION: 'v{{buildbot_version}}'
-  PTK_URI: '{{asset_host}}/download/610-009981-015_SW_PTK_5.3_Client_RevA.tar'
+  PTK_URI: '{{asset_downloads}}/610-009981-015_SW_PTK_5.3_Client_RevA.tar'
   PTK_HSM_HOST: '{{hsm_host}}'
 
 # directory of the image source
diff --git a/roles/buildbot-worker-ntel/templates/ptxdistrc.j2 b/roles/buildbot-worker-ntel/templates/ptxdistrc.j2
deleted file mode 100644
index 4d2b48e91..000000000
--- a/roles/buildbot-worker-ntel/templates/ptxdistrc.j2
+++ /dev/null
@@ -1,76 +0,0 @@
-#
-# Automatically generated file; DO NOT EDIT.
-# PTXdist 2012.09.1-00373-gee9622f
-#
-
-#
-# PTXDIST Setup Menu
-#
-
-#
-# User
-#
-PTXCONF_SETUP_USER_NAME="NovaTech Buildbot"
-PTXCONF_SETUP_USER_EMAIL="buildbot@novatechweb.com"
-
-#
-# Proxies
-#
-PTXCONF_SETUP_FTP_PROXY=""
-PTXCONF_SETUP_HTTP_PROXY=""
-PTXCONF_SETUP_HTTPS_PROXY=""
-PTXCONF_SETUP_NO_PROXY=""
-
-#
-# Project Searchpath
-#
-PTXCONF_SETUP_PROJECTPATH="${PTXDIST_TOPDIR}/projects"
-
-#
-# Source Directories
-#
-PTXCONF_SETUP_SRCDIR="{{cache_path}}/downloads"
-
-#
-# Source Download
-#
-# PTXCONF_SETUP_NO_DOWNLOAD is not set
-# PTXCONF_SETUP_PTXMIRROR_ONLY is not set
-PTXCONF_SETUP_PTXMIRROR="{{asset_host}}/orphans"
-PTXCONF_SETUP_DEBMIRROR="{{asset_host}}/debian"
-PTXCONF_SETUP_SFMIRROR="{{asset_host}}/sourceforge"
-PTXCONF_SETUP_GNUMIRROR="{{asset_host}}/gnu"
-PTXCONF_SETUP_XORGMIRROR="{{asset_host}}/xorg"
-PTXCONF_SETUP_KERNELMIRROR="{{asset_host}}/kernel"
-PTXCONF_SETUP_PYPIMIRROR="{{asset_host}}/pypi"
-# PTXCONF_SETUP_CHECK_ALWAYS is not set
-PTXCONF_SETUP_CHECK_NOTEMPTY=y
-# PTXCONF_SETUP_CHECK_NEVER is not set
-PTXCONF_SETUP_CHECK="notempty"
-
-#
-# IPKG Repository
-#
-PTXCONF_SETUP_IPKG_REPOSITORY="{{cache_path}}/ipkg-repository"
-
-#
-# Java SDK
-#
-PTXCONF_SETUP_JAVA_SDK="/usr"
-
-#
-# Developer Options
-#
-# PTXCONF_SETUP_DISABLE_LOCAL_CHECK is not set
-PTXCONF_SETUP_ENV_WHITELIST=""
-# PTXCONF_SETUP_COMMON_CACHE is not set
-# PTXCONF_SETUP_GEN_DEP_TREE is not set
-# PTXCONF_SETUP_CHECK_EXIT_ON_ERROR is not set
-PTXCONF_SETUP_CCACHE=y
-PTXCONF_SETUP_PATCHIN_GIT=y
-# PTXCONF_SETUP_NFS_REL_SYMLINK is not set
-# PTXCONF_SETUP_DIRECT_CLEAN is not set
-PTXCONF_SETUP_HOST_CPP="cpp"
-PTXCONF_SETUP_HOST_CC="gcc"
-PTXCONF_SETUP_HOST_CXX="g++"
-PTXCONF_SETUP_HOST_MAKE="make"
diff --git a/roles/buildbot-worker-ntel/templates/pydistutils.cfg.j2 b/roles/buildbot-worker-ntel/templates/pydistutils.cfg.j2
index 1d205ea20..8eeecd74b 100644
--- a/roles/buildbot-worker-ntel/templates/pydistutils.cfg.j2
+++ b/roles/buildbot-worker-ntel/templates/pydistutils.cfg.j2
@@ -1,2 +1,2 @@
 [easy_install]
-index_url = {{asset_host}}/pypi/simple
+index_url = {{asset_pypi}}/simple
diff --git a/roles/buildbot-worker-ptxdist/defaults/main.yml b/roles/buildbot-worker-ptxdist/defaults/main.yml
index 5a128da14..ebe2d7edb 100644
--- a/roles/buildbot-worker-ptxdist/defaults/main.yml
+++ b/roles/buildbot-worker-ptxdist/defaults/main.yml
@@ -27,7 +27,6 @@ cache_volume: '{{ docker_name_prefix }}{{ container_hostname }}-cache'
 
 # environment passed to the container
 container_env:
-  ASSET_HOST: '{{asset_host}}/ptxdist'
   BUILDBOT_KNOWN_HOSTS_FILE: '{{secrets_path}}/{{secrets_known_hosts}}'
   BUILDBOT_MASTER: '{{ buildbot_master }}'
   BUILDBOT_WORKER_PORT: '{{ buildbot_worker_port }}'
@@ -43,9 +42,6 @@ container_name: '{{ docker_name_prefix }}{{ container_hostname }}'
 # network configuration for the confainer
 container_networks: []
 
-# network configuration for the confainer
-container_networks: []
-
 # port configuration of the container
 container_port_args: []
 
diff --git a/roles/buildbot-worker-ptxdist/templates/ptxdistrc.j2 b/roles/buildbot-worker-ptxdist/templates/ptxdistrc.j2
index 4d2b48e91..d401f2c2e 100644
--- a/roles/buildbot-worker-ptxdist/templates/ptxdistrc.j2
+++ b/roles/buildbot-worker-ptxdist/templates/ptxdistrc.j2
@@ -36,13 +36,13 @@ PTXCONF_SETUP_SRCDIR="{{cache_path}}/downloads"
 #
 # PTXCONF_SETUP_NO_DOWNLOAD is not set
 # PTXCONF_SETUP_PTXMIRROR_ONLY is not set
-PTXCONF_SETUP_PTXMIRROR="{{asset_host}}/orphans"
-PTXCONF_SETUP_DEBMIRROR="{{asset_host}}/debian"
-PTXCONF_SETUP_SFMIRROR="{{asset_host}}/sourceforge"
-PTXCONF_SETUP_GNUMIRROR="{{asset_host}}/gnu"
-PTXCONF_SETUP_XORGMIRROR="{{asset_host}}/xorg"
-PTXCONF_SETUP_KERNELMIRROR="{{asset_host}}/kernel"
-PTXCONF_SETUP_PYPIMIRROR="{{asset_host}}/pypi"
+PTXCONF_SETUP_PTXMIRROR="{{asset_orphans}}"
+PTXCONF_SETUP_DEBMIRROR="{{asset_debian}}"
+PTXCONF_SETUP_SFMIRROR="{{asset_sourceforge}}"
+PTXCONF_SETUP_GNUMIRROR="{{asset_gnu}}"
+PTXCONF_SETUP_XORGMIRROR="{{asset_xorg}}"
+PTXCONF_SETUP_KERNELMIRROR="{{asset_kernel}}"
+PTXCONF_SETUP_PYPIMIRROR="{{asset_pypi}}"
 # PTXCONF_SETUP_CHECK_ALWAYS is not set
 PTXCONF_SETUP_CHECK_NOTEMPTY=y
 # PTXCONF_SETUP_CHECK_NEVER is not set
diff --git a/roles/buildbot-worker-ptxdist/templates/pydistutils.cfg.j2 b/roles/buildbot-worker-ptxdist/templates/pydistutils.cfg.j2
index 1d205ea20..8eeecd74b 100644
--- a/roles/buildbot-worker-ptxdist/templates/pydistutils.cfg.j2
+++ b/roles/buildbot-worker-ptxdist/templates/pydistutils.cfg.j2
@@ -1,2 +1,2 @@
 [easy_install]
-index_url = {{asset_host}}/pypi/simple
+index_url = {{asset_pypi}}/simple
